BRAKER with long read RNA-Seq (only) and protein data

We forked GeneMark-ETP and hard coded that StringTie will perform long read assembly in that particular version. If you want to use this 'fast-hack' version for BRAKER, you have to prepare the BAM file with long read to genome spliced alignments outside of BRAKER, e.g.:

```bash
T=48 # adapt to your number of threads
minimap2 -t${T} -ax splice:hq -uf genome.fa isoseq.fa > isoseq.sam     
samtools view -bS --threads ${T} isoseq.sam -o isoseq.bam
```

Pull the adapted container:
```
singularity build braker3_lr.sif docker://teambraker/braker3:isoseq
```

Calling BRAKER3 with a BAM file of spliced-aligned IsoSeq Reads:
```bash
singularity exec -B ${PWD}:${PWD} braker3_lr.sif braker.pl --genome=genome.fa --prot_seq=protein_db.fa â€“-bam=isoseq.bam --threads=${T} 
```

> Warning Do NOT mix short read and long read data in this BRAKER/GeneMark-ETP variant!

> Warning The accuracy of gene prediction here heavily depends on the depth of your isoseq data. We verified with PacBio HiFi reads from 2022 that given sufficient completeness of the assembled transcriptome you will reach similar results as with short reads. However, we also observed a drop in accuracy compared to short reads when using other long read data sets with higher error rates and less sequencing depth.

```bash
singularity exec -B ${PWD}:${PWD} ${BRAKER_LR_SIF} braker.pl
        --genome=1.references/regina/Regina_assembly.chr.softmasked.fasta
        --prot_seq=db/Viridiplantae.fa --bam=0.bams/regina/Regina_bud_aln.sorted.bam
        --workingdir=${wd}
        --threads=${T}
        --busco_lineage embryophyta_odb10 
```
