# BRAKER with long read RNA-Seq (only) and protein data
We forked GeneMark-ETP and hard coded that StringTie will perform long read assembly in that particular version. If you want to use this 'fast-hack' version for BRAKER, you have to prepare the BAM file with long read to genome spliced alignments outside of BRAKER, e.g.:

1. Get FASTA file from clustered flnc sequences from Iso-seq pipeline
```bash
samtools fasta ../Regina_bud_clustered.sorted.bam > Regina_bud.fasta
```

2. Align sequences against genome assembly
```bash
T=48 # adapt to your number of threads
minimap2 -t${T} -ax splice:hq -uf Regina_assembly.chr.fasta Regina_bud.fasta > Regina_isoforms_aln.sam
samtools view -bS Regina_bud_aln.sam > Regina_bud_aln.bam
samtools sort -o Regina_bud_aln.sorted.bam Regina_bud_aln.bam
```

3. Calling BRAKER3 with a BAM file of spliced-aligned IsoSeq Reads:
```bash
# singularity exec -B ${PWD}:${PWD} braker3_lr.sif braker.pl --genome=genome.fa --prot_seq=protein_db.fa â€“-bam=isoseq.bam --threads=${T} 

singularity exec -B ${PWD}:${PWD} ${BRAKER_LR_SIF} braker.pl \
        --genome=1.references/regina/Regina_assembly.chr.softmasked.fasta \
        --prot_seq=db/Viridiplantae.fa \
        --bam=0.bams/regina/Regina_bud_aln.sorted.bam \
        --workingdir=${wd} \
        --threads=${T} \
        --busco_lineage embryophyta_odb10 
```
* Input:
  - `Regina_assembly.chr.softmasked.fasta`: Genome assembly softmasked
  - `Regina_bud_aln.sorted.bam`: bam file with spliced alignments from Iso-seq clustered FLNC sequences.
  - `Viridiplantae.fa`: [OrthoDB_v12](https://bioinf.uni-greifswald.de/bioinf/partitioned_odb12/) plant partition used to generate protein hints.
  - `embryophyta_odb10`: BUSCO lineage, BRAKER will run compleasm on genome level to generate hints from BUSCO to enhance BUSCO discovery in the protein set.

> **Warning** Do NOT mix short read and long read data in this BRAKER/GeneMark-ETP variant!

> **Warning** The accuracy of gene prediction here heavily depends on the depth of your isoseq data. We verified with PacBio HiFi reads from 2022 that given sufficient completeness of the assembled transcriptome you will reach similar results as with short reads. However, we also observed a drop in accuracy compared to short reads when using other long read data sets with higher error rates and less sequencing depth.

4. Once the annotation is done for each tissue we have to merge them into one transcriptome using `tama_merge.py`. First, gtf files need to be converted into bed files:
```bash
gtf_to_bed Regina_bud_braker.gtf Regina_bud.bed
gtf_to_bed Regina_leaf_braker.gtf Regina_leaf.bed
```
```bash
awk 'BEGIN{OFS="\t"} {split($4, a, "."); $4 = a[1]";"$4; print}' Regina_bud.bed > Regina_bud_fixed.bed
awk 'BEGIN{OFS="\t"} {split($4, a, "."); $4 = a[1]";"$4; print}' Regina_leaf.bed > Regina_leaf_fixed.bed
```

5. Then, filelist.txt file should be created using the following format:
```bash
Regina_bud_fixed.bed    no_cap  1,1,1   bud
Regina_leaf_fixed.bed   no_cap  1,1,1   leaf
```

6. Finally, merge two transcriptomes into one:
```bash
tama_merge.py -f filelist.txt -d merge_dup -p bud_leaf
```
